driver/configuration/provider_viper.go

```diff
@@ -197,7 +204,53 @@ func (v *ViperProvider) pipelineIsEnabled(prefix, id string) bool {
 	return viperx.GetBool(v.l, fmt.Sprintf("%s.%s.enabled", prefix, id), false)
 }
 
+func (v *ViperProvider) hashPipelineConfig(prefix, id string, override json.RawMessage) (uint64, error) {
+	ts := viper.ConfigChangeAt().UnixNano()
+	b := make([]byte, 8)
+	binary.LittleEndian.PutUint64(b, uint64(ts))
+
+	env, err := json.Marshal(os.Environ())
+	if err != nil {
+		return 0, err
+	}
+
+	slices := [][]byte{
+		[]byte(prefix),
+		[]byte(id),
+		[]byte(override),
+		[]byte(b),
+		[]byte(env),
+	}
+
+	var hashSlices []byte
+	for _, s := range slices {
+		hashSlices = append(hashSlices, s...)
+	}
+
+	return crc64.Checksum(hashSlices, crc64.MakeTable(crc64.ECMA)), nil
+}
+
 func (v *ViperProvider) PipelineConfig(prefix, id string, override json.RawMessage, dest interface{}) error {
+	hash, err := v.hashPipelineConfig(prefix, id, override)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	c, ok := v.configCachge[hash]
+	if ok {
+		if dest != nil {
+			marshalled, err := json.Marshal(c)
```

Comment:
Let's safe one round trip of encoding here - see line https://github.com/ory/oathkeeper/pull/348/files#diff-2449a6ea083767b149400840c05f41bdR308

I thought it might be a good idea to use glob, so I wrote a small benchmark. Unfortunately, glob seems to underperform when having to deal with interfaces (**so let's stick to json**), here are the results:

```
go test -bench . -benchmem
goos: darwin
goarch: amd64
pkg: github.com/ory/oathkeeper/docs
BenchmarkGobJSON/gob-12                     7848            151588 ns/op           55343 B/op       1234 allocs/op
BenchmarkGobJSON/json-12                   20253             57361 ns/op           32480 B/op        489 allocs/op
```

```go
package docs

import (
	"bytes"
	"encoding/gob"
	"encoding/json"
	"io/ioutil"
	"testing"

	"github.com/ghodss/yaml"
	"github.com/stretchr/testify/require"
)

func BenchmarkGobJSON(b *testing.B) {
	gob.Register(map[string]interface{}{})
	gob.Register([]interface{}{})

	c, err := ioutil.ReadFile(".oathkeeper.yaml")
	require.NoError(b, err)

	conf, err := yaml.YAMLToJSON(c)
	require.NoError(b, err)

	var o map[string]interface{}
	require.NoError(b, json.NewDecoder(bytes.NewBuffer(conf)).Decode(&o))

	var gb bytes.Buffer
	require.NoError(b, gob.NewEncoder(&gb).Encode(&o))

	b.Run("gob", func(b *testing.B) {
		for i := 0; i < b.N; i++ {
			b.StopTimer()
			m := make(map[string]interface{})
			buf := bytes.NewBuffer(gb.Bytes())
			b.StartTimer()

			require.NoError(b, gob.NewDecoder(buf).Decode(&m))
		}
	})

	b.Run("json", func(b *testing.B) {
		for i := 0; i < b.N; i++ {
			b.StopTimer()
			m := make(map[string]interface{})
			buf := bytes.NewBuffer(conf)
			b.StartTimer()

			require.NoError(b, json.NewDecoder(buf).Decode(&m))
		}
	})
}
```